### PR TITLE
Allow a branch to `export IS_RELEASE=0` to force-publish

### PR DESCRIPTION
**Pull request by lancelet**
Ping @stephanh, @SamRoberts.

  - Currently, there is a hard-coded list of branches that are
    published (see settings.sh). This change allows a .travis.yml
    file to `export IS_RELEASE=0;`, which would then force that
    branch to be published.
  - This is a possible solution to Issue #11. Please feel free to
    comment / suggest an alternative approach. Right now, it's
    just a very minimal change to an existing variable that is
    currently used for this purpose.

### CODE REVIEW

**Comment by lancelet** [comment]
BTW... I think this is a horrible approach; I'd rather have a command-line flag. However, I've never written a command line parser in `bash`, and I *really* don't want to waste any time on this...

File: sbt-deploy-to.sh [hunk 1]
```diff
@@ -33,10 +37,16 @@ readonly location="$( cd $(dirname $0) && pwd -P )"
 
 source $location/settings.sh
 
-set +e
-isReleaseBranch $TRAVIS_BRANCH
-IS_RELEASE=$?
-set -e
+# only override the IS_RELEASE variable if it is not already set to zero
+if [ $IS_RELEASE -ne 0 ]; then
+    echo "IS_RELEASE was NOT set to 0 - checking whether to publish"
```
Comment (SamRoberts) [inline sbt-deploy-to.sh]: if we output this message, probably also a good idea to output whether we decided to publish or not

**Comment by SamRoberts** [comment]
Hrmm. My only problem with this is that it encourages you to make a change to `.travis.yml` which you really should remember to remove before merging to master (or should you?), which I think will often be forgotten.

I do like the ability to specify whether the branch is release branch in my own repo though.

I think a better approach would be to specifiy a list of project specific release branches in the `.travis.yml`, which gets appended to the master list in `ci`. That way it's ok (and in fact even encouraged) to push the change to master, and each project can end up with a handful of project-specific release branches which they can re-use whenever appropriate.

**Comment by lancelet** [comment]
@SamRoberts ah - good idea. I'll take a look at setting up something like that next week.

**Comment by luke-clifton** [comment]
Wouldn't it be better to track this outside of the repository itself... doesn't really seem like it should belong in there (why do I need to touch repository to get it released). Travis lets you set per project environment variables. I propose that we use such an environment variable to extend the list of branches that are to be released.

Pros:
 * This lets you see in one place which branches are getting released (rather than having to look in every branch)
 * I can momentarily add a branch, rebuild it, and remove it again to get a point release for a particular branch (which, given the motivation for this change, a demo, seems like a good feature)
 * Don't dirty the history with "meaningless" commits to toggle releases on and off.
 * No need to worry about changing things before merging with master

Cons:
 * Not in source control could be considered a con?

**Comment by stephanh** [comment]
@luke-clifton when we first did this travis did not have that feature. I think it would be good to take advantage of that feature now.

**Comment by lancelet** [comment]
Will close this PR and re-examine using an environment variable set in Travis to identify which branches should be published.

**Pull request closed without merging**